ci/system_test/run.py

```diff
@@ -0,0 +1,303 @@
+            command=command,
+            args=args,
+            stdin=stdin,
+            workdir=workdir,
+        )
+        command = f"cd {workdir}; " + command
+        if args:
+            command += " " + " ".join(args)
+        try:
+            stdin_stream, stdout_stream, stderr_stream = self._ssh_client.exec_command(
+                command
+            )
+
+            if stdin:
+                stdin_stream.write(stdin)
+                stdin_stream.close()
+
+            if live:
+                while True:
+                    line = stdout_stream.readline()
+                    stdout += line
+                    if not line:
+                        break
+                    print(line, end="")
+            else:
+                stdout = stdout_stream.read()
+
+            stderr = stderr_stream.read()
+
+            exit_status = stdout_stream.channel.recv_exit_status()
+            if exit_status != 0 and not suppress_errors:
+                raise RuntimeError(f"Command failed with exit status: {exit_status}")
+        except (paramiko.SSHException, RuntimeError) as e:
+            self._logger.error(
+                "Failed running command on data cluster",
+                command=command,
+                error=e,
+                stdout=stdout,
+                stderr=stderr,
+                exit_status=exit_status,
+            )
+            raise
+        else:
+            self._logger.debug(
+                "Successfully ran command on data cluster",
+                command=command,
+                stdout=stdout,
+                stderr=stderr,
+                exit_status=exit_status,
+            )
+            return stdout
+
+    def _get_provctl_version_and_url(self):
+        response = requests.get(
+            self.Constants.provctl_releases,
+            params={"access_token": self._github_access_key},
+        )
+        response.raise_for_status()
+        latest_provazio_release = json.loads(response.content)
+        for asset in latest_provazio_release["assets"]:
+            if asset["name"] == self.Constants.provctl_binary_format.format(
+                release_name=latest_provazio_release["name"]
+            ):
+                self._logger.debug(
+                    "Got provactl release url",
+                    release=latest_provazio_release["name"],
+                    name=asset["name"],
+                    url=asset["url"],
+                )
+                return asset["name"], asset["url"]
+
+        raise RuntimeError("provctl binary not found")
+
+    def _prepare_test_env(self, mlrun_path):
+
+        self._logger.debug("Cloning mlrun from github")
+        self._run_command(
+            "git", args=["clone", self.Constants.git_url, str(mlrun_path)]
+        )
+
+        self._logger.debug("Populating system tests env.yml")
+        self._run_command(
+            "cat > ",
+            args=[str(mlrun_path / self.Constants.system_tests_env_yaml)],
```

Comment:
If I'm understanding correctly `mlrun_path` is `/home/iguazio/mlrun-ci/mlrun` and `self.Constants.system_tests_env_yaml` is `/home/iguazio/mlrun-ci/mlrun/tests/system/env.yml`
so how this works ?

Reply:
just understood that those are `Path` objects so it does correctly resolves to  `/home/iguazio/mlrun-ci/mlrun/tests/system/env.yml` but the `mlrun_path` is not needed here